From 62d8a79156a62cdb4e44c1cd4bdb68d6ecbef19b Mon Sep 17 00:00:00 2001
From: Lei Qian <lei.qian@amlogic.com>
Date: Tue, 23 Oct 2018 17:18:54 +0800
Subject: [PATCH] WindowState: fixed OverlayView can't display issue [1/1]

PD#SWPL-745

Problem:
OverlayView can't display

Solotion:
setRelativeLayer for TYPE_APPLICATION_MEDIA_OVERLAY

Verify:
verify by einstein

Change-Id: I4828e60d7c915076013fb2b5666512a89416b1b7
Signed-off-by: Lei Qian <lei.qian@amlogic.com>
---
 services/core/java/com/android/server/tv/TvInputManagerService.java | 4 ++++
 services/core/java/com/android/server/wm/WindowState.java           | 6 +++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/tv/TvInputManagerService.java b/services/core/java/com/android/server/tv/TvInputManagerService.java
index d5e59c8..d104488 100644
--- a/services/core/java/com/android/server/tv/TvInputManagerService.java
+++ b/services/core/java/com/android/server/tv/TvInputManagerService.java
@@ -827,6 +827,10 @@ public final class TvInputManagerService extends SystemService {
     private void setStateLocked(String inputId, int state, int userId) {
         UserState userState = getOrCreateUserStateLocked(userId);
         TvInputState inputState = userState.inputMap.get(inputId);
+        if (inputState == null ||inputState.info == null) {
+            Slog.d(TAG,"Can't get input info,exit!!!!!");
+            return;
+        }
         ServiceState serviceState = userState.serviceStateMap.get(inputState.info.getComponent());
         int oldState = inputState.state;
         inputState.state = state;
diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index bee70a0..4257c37 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -4797,7 +4797,11 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
             if (w.mAttrs.type == TYPE_APPLICATION_MEDIA) {
                 w.assignLayer(t, -2);
             } else if (w.mAttrs.type == TYPE_APPLICATION_MEDIA_OVERLAY) {
-                w.assignLayer(t, -1);
+                if (mWinAnimator.hasSurface()) {
+                    w.assignRelativeLayer(t, mWinAnimator.mSurfaceController.mSurfaceControl, -1);
+                } else {
+                    w.assignLayer(t, -1);
+                }
             } else {
                 w.assignLayer(t, layer);
             }
-- 
1.9.1

